File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
       String num=String.valueOf(x);
       int left=0;
       int right=num.length()-1;
       if(x<0){
        return false;
       }
       while(left<right){
        if(num.charAt(left)!=num.charAt(right))
        {
            return false;
        }
        left++;
        right--;
          
    }
       return true;
    }
}

File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        string num=to_string(x);
        
        int start=0;
        int end=num.length()-1;
        
        while(start<end){
            if(num.at(start)!=num.at(end)){
                return false;
            }
        start++;
        end--;
        }
        
        return true;
        
    }
};